pkg/vpcmodel/nodesExplainability.go

```diff
@@ -56,66 +158,141 @@ func (c *VPCConfig) getFiltersEnablingRulesBetweenNodesPerDirectionAndLayer(
+		noConnection := fmt.Sprintf("No connection between %v and %v;", src.Name(), dst.Name())
+		switch {
+		case noIngressRules && noEgressRules:
+			resStr += fmt.Sprintf("%v connection blocked both by ingress and egress\n", noConnection)
+		case noIngressRules:
+			resStr += fmt.Sprintf("%v connection blocked by ingress\n", noConnection)
+			if needEgress {
+				resStr += egressRulesStr
+			}
+		case noEgressRules:
+			resStr += fmt.Sprintf("%v connection blocked by egress\n", noConnection)
+			if needIngress {
+				resStr += ingressRulesStr
+			}
+		default: // there is a connection
+			resStr = fmt.Sprintf("The following connection exists between %v and %v: %v; its enabled by\n", src.Name(), dst.Name(),
+				rulesSrcDst.conn.String())
+			if needEgress {
+				resStr += egressRulesStr
+			}
+			if needIngress {
+				resStr += ingressRulesStr
+			}
 		}
-		conn, ok := connectivity.AllowedConnsCombined[src][dst]
-		if !ok {
-			return "", fmt.Errorf("error: there is a connection between %v and %v, but connection computation failed",
-				src.Name(), dst.Name())
+		// todo: not adding newlines in between since this will be replaced by grouping in the next PR
+	}
+	return resStr, nil
+}
+
+// todo: connectivity is computed for the entire network, even though we need only for specific src, dst pairs
+// this is seems the time spent here should be neglectable, not worth the effort of adding dedicated code.
+func (explanationStruct *explainStruct) computeConnections(c *VPCConfig) error {
+	connectivity, err := c.GetVPCNetworkConnectivity(false) // computes connectivity
+	if err != nil {
+		return err
+	}
+	for _, rulesSrcDst := range *explanationStruct {
+		// is there a connection?
+		if (len(rulesSrcDst.rules.egressRules) > 0 || !rulesSrcDst.src.IsInternal()) && // egress enabled or not needed
+			(len(rulesSrcDst.rules.ingressRules) > 0 || !rulesSrcDst.dst.IsInternal()) { // ingress enabled or not needed
+			conn, err := connectivity.getConnection(c, rulesSrcDst.src, rulesSrcDst.dst)
+			if err != nil {
+				return err
+			}
+			rulesSrcDst.conn = conn
 		}
-		return fmt.Sprintf("The following connection exists between %v and %v: %v; its enabled by\nEgress Rules:\n~~~~~~~~~~~~~\n%v\n"+
-			"Ingress Rules:\n~~~~~~~~~~~~~~\n%v\n", src.Name(), dst.Name(), conn.String(), egressRulesStr, ingressRulesStr), nil
 	}
+	return nil
+}
+
+// given that there is a connection between src to dst, gets it
+// if src or dst is a node then the node is from getCidrExternalNodes,
+// thus there is a node in VPCConfig that either equal to or contains it.
+func (v *VPCConnectivity) getConnection(c *VPCConfig, src, dst Node) (conn *common.ConnectionSet, err error) {
+	srcForConnection, err1 := c.getContainingConfigNode(src)
+	if err1 != nil {
+		return nil, err1
+	}
+	dstForConnection, err2 := c.getContainingConfigNode(dst)
+	if err2 != nil {
+		return nil, err2
+	}
+	conn, ok := v.AllowedConnsCombined[srcForConnection][dstForConnection]
```

Comment:
shouldn't first check that `v.AllowedConnsCombined[srcForConnection]` is ok, before accessing it?

Reply:
Yep, seems I should. Done.